src/Catty/DataModel/VariablesContainer/UserDataContainer.swift

```diff
@@ -20,7 +20,128 @@
  *  along with this program.  If not, see http://www.gnu.org/licenses/.
  */
 
-@objc extension UserDataContainer {
+@objc(UserDataContainer)
+@objcMembers class UserDataContainer: NSObject {
+    private var _variables: SynchronizedArray<UserVariable>
+    private var _lists: SynchronizedArray<UserList>
+
+    override init() {
+        self._variables = SynchronizedArray<UserVariable>()
+        self._lists = SynchronizedArray<UserList>()
+    }
+
+    func variables() -> [UserVariable] {
+        var variables = [UserVariable]()
+        for index in 0..<_variables.count {
+            if let variable = _variables[index] {
+                variables.append(variable)
+            }
+        }
+        return variables
+    }
+
+    func lists() -> [UserList] {
+        var lists = [UserList]()
+        for index in 0..<_lists.count {
+            if let list = _lists[index] {
+                lists.append(list)
+            }
+        }
+        return lists
+    }
+
+    func removeAllVariables() {
+        _variables.removeAll()
+    }
+
+    func removeAllLists() {
+        _lists.removeAll()
+    }
+
+    func getUserVariable(withName name: String) -> UserVariable? {
+        for index in 0..<_variables.count {
+            if let variable = _variables[index], variable.name == name {
+                return variable
+            }
+        }
+        return nil
+    }
+
+    func getUserList(withName name: String) -> UserList? {
+        for index in 0..<_lists.count {
+            if let list = _lists[index], list.name == name {
+                return list
+            }
+        }
+        return nil
+    }
+
+    func removeUserVariable(withName name: String) -> Bool {
+        if let _ = getUserVariable(withName: name) {
+            _variables.remove(name: name)
+            return true
+        }
+        return false
+    }
+
+    func removeUserList(withName name: String) -> Bool {
+        if let _ = getUserList(withName: name) {
+            _lists.remove(name: name)
+            return true
+        }
+        return false
+    }
+
+    @objc(containsVariable:)
+    func contains(_ variable: UserVariable) -> Bool {
+        _variables.contains(variable)
+    }
+
+    @objc(containsList:)
+    func contains(_ list: UserList) -> Bool {
+        _lists.contains(list)
+    }
+
+    @objc(addVariable:)
+    func add(_ variable: UserVariable) {
+        if !_variables.contains(variable) {
+            _variables.append(variable)
+        }
+    }
+
+    @objc(addList:)
+    func add(_ list: UserList) {
+        if !_lists.contains(list) {
+            _lists.append(list)
+        }
+    }
+
+    func isEqual(to userDataContainer: UserDataContainer) -> Bool {
```

Comment:
Please override the generic method: `override func isEqual(_ object: Any?) -> Bool`